agent/lib/kontena/workers/service_pod_worker.rb

```diff
@@ -37,6 +39,10 @@ def ensure_desired_state
       exclusive {
         debug "state of #{service_pod.name}: #{service_pod.desired_state}"
         service_container = get_container(service_pod.service_id, service_pod.instance_number)
+        if service_container && !service_pod.running?
```

Comment:
I think it makes sense to keep the `stop_health_check if !service_pod.running?` and `ensure_health_check if service_pod.running?` logic here.

Reason being: if we are stopping a container, then the health check worker must be terminated before we do that.